samples/Config.CustomConfigurationProviders.Sample/Microsoft.Extensions.Configuration.ConfigFile/ConfigFileConfigurationProviderExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Extensions.Configuration.ConfigFile
+{
+    public static class ConfigFileConfigurationProviderExtensions
+    {
+        /// <summary>
+        /// Adds configuration values of a *.config file to the ConfigurationBuilder
+        /// </summary>
+        /// <param name="builder">Builder to add configuration values to</param>
+        /// <param name="configContents">Contents of *.config file</param>
+        /// <param name="parsers">Additional parsers to use to parse the config contents</param>
+        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder builder, string configContents, params IConfigurationParser[] parsers)
+        {
+            if (configContents == null)
+            {
+                throw new ArgumentNullException(nameof(configContents));
+            }
+            else if (string.IsNullOrEmpty(configContents))
+            {
+                throw new ArgumentException(nameof(configContents), "Contents for configuration cannot be empty!");
```

Comment:
The order of these two parameters is incorrect (need to fix in a few places in this PR). https://msdn.microsoft.com/en-us/library/sxykka64(v=vs.110).aspx